client/src/main.jsx

```diff
@@ -200,6 +207,21 @@ const router = createBrowserRouter([
         }),
         action: async ({ request, params }) => {
           const formData = await request.formData();
+          const formDataObject = {
+            name: formData.get("name"),
+            amount: parseInt(formData.get("amount"), 10),
+            startDate: formData.get("start_date"),
+            endDate: formData.get("end_date"),
+            categoryId: parseInt(formData.get("category"), 10),
+          };
+
+          const validatedData = budgetFormValidator({
+            formData: formDataObject,
+          });
```

Comment:
**question**: What's the point of putting formDataObject inside another object with only one attribute ?

Reply:
Question discussed in person